### PR TITLE
Cleanup references to sample config in the docs and redirect users to configuration manual

### PR DESCRIPTION
**Pull request by H-Shay**
Now that the comments in the configuration file have been removed, users must use the configuration manual instead of referencing the comments in the sample config. This PR removes relevant references to the sample config in the docs and adds references to the configuration manual instead.

### CODE REVIEW

File: docs/admin_api/user_admin_api.md [hunk 1]
```diff
@@ -124,9 +124,9 @@ Body parameters:
   - `address` - string. Value of third-party ID.
   belonging to a user.
 - `external_ids` - array, optional. Allow setting the identifier of the external identity
-  provider for SSO (Single sign-on). Details in
-  [Sample Configuration File](../usage/configuration/homeserver_sample_config.html)
-  section `sso` and `oidc_providers`.
+  provider for SSO (Single sign-on). Details in the
+  [configuration manual](../usage/configuration/config_documentation.md) under the
+  sections `sso` and `oidc_providers`.
```
Comment (reivilibre) [inline docs/admin_api/user_admin_api.md:129]: is this one linkable?

File: docs/code_style.md [hunk 2]
```diff
@@ -70,80 +70,59 @@ on save as they take a while and can be very resource intensive.
...
 
--   Sections should be separated with a heading consisting of a single
-    line prefixed and suffixed with `##`. There should be **two** blank
-    lines before the section header, and **one** after.
--   Each option should be listed in the file with the following format:
-    -   A comment describing the setting. Each line of this comment
-        should be prefixed with a hash (`#`) and a space.
+- Each option should be listed in the config manual with the following format:
+      
+    - The name of the option, prefixed by "###".
```
Comment (reivilibre) [inline docs/code_style.md:86]:
```suggestion
    - The name of the option, prefixed by `###`.
```

File: docs/code_style.md [hunk 3]
```diff
@@ -70,80 +70,59 @@ on save as they take a while and can be very resource intensive.
...
-    lines before the section header, and **one** after.
--   Each option should be listed in the file with the following format:
-    -   A comment describing the setting. Each line of this comment
-        should be prefixed with a hash (`#`) and a space.
+- Each option should be listed in the config manual with the following format:
+      
+    - The name of the option, prefixed by "###". 
 
-        The comment should describe the default behaviour (ie, what
+    - A comment which describes the default behaviour (ie, what
```
Comment (reivilibre) [inline docs/code_style.md:88]:
```suggestion
    - A comment which describes the default behaviour (i.e. what
```

File: docs/code_style.md [hunk 4]
```diff
@@ -70,80 +70,59 @@ on save as they take a while and can be very resource intensive.
...
--   There should be a blank line between each option.
--   Where several settings are grouped into a single dict, *avoid* the
-    convention where the whole block is commented out, resulting in
-    comment lines starting `# #`, as this is hard to read and confusing
-    to edit. Instead, leave the top-level config option uncommented, and
-    follow the conventions above for sub-options. Ensure that your code
+        should be the *opposite* to the default. For other options, the example should give
+        some non-default value which is likely to be useful to the reader.
+
+- There should be a line between each option, which can be achieved by adding --- before and
```
Comment (reivilibre) [inline docs/code_style.md:97]:
```suggestion
- There should be a horizontal rule between each option, which can be achieved by adding `---` before and
```

File: docs/code_style.md [hunk 5]
```diff
@@ -70,80 +70,59 @@ on save as they take a while and can be very resource intensive.
...
+  after the option.
+- Where several settings are grouped into a single dict, ensure that your code
```
Comment (reivilibre) [inline docs/code_style.md:99]: I don't think this statement is related to the config manual but rather the implementation, so probably separate it out into a new paragraph of some kind

File: docs/code_style.md [hunk 6]
```diff
@@ -70,80 +70,59 @@ on save as they take a while and can be very resource intensive.
...
+- `true` and `false` are spelt thus (as opposed to `True`, etc.)
 
 Example:
 
+---
+### `modules`
+
+Use the `module` sub-option to add modules under this option to extend functionality. 
+The `module` setting then has a sub-option, `config`, which can be used to define some configuration
+for the `module`.
```
Comment (reivilibre) [inline docs/code_style.md:111]: I find this paragraph quite hard to follow (I realise this is just an example)

File: docs/message_retention_policies.md [hunk 7]
```diff
@@ -65,8 +65,8 @@ message retention policy configured in its state. This allows server
 admins to ensure that messages are never kept indefinitely in a server's
 database. 
 
-A default policy can be defined as such, in the `retention` section of
-the configuration file:
+A default policy can be defined as such, by adding the `retention` option in
+the configuration file and adding these sub-options:
```
Comment (reivilibre) [inline docs/message_retention_policies.md:69]: is calling sections 'options' and the options within 'sub-options' something we've already started doing?
I think I'm partial to calling them sections, but I'm not going to make a fuss of it either, it's probably just what I'm used to.

File: docs/usage/configuration/user_authentication/single_sign_on/cas.md [hunk 8]
```diff
@@ -4,5 +4,4 @@ Synapse supports authenticating users via the [Central Authentication
 Service protocol](https://en.wikipedia.org/wiki/Central_Authentication_Service)
 (CAS) natively.
 
-Please see the `cas_config` and `sso` sections of the [Synapse configuration
-file](../../../configuration/homeserver_sample_config.md) for more details.
\ No newline at end of file
+Please see the `cas_config` and `sso` sections of the [configuration manual](../../../configuration/config_documentation.md) for more details.
```
Comment (reivilibre) [inline docs/usage/configuration/user_authentication/single_sign_on/cas.md:7]: please deep link if possible :) (since we actually can with this new manual!)

File: changelog.d/13077.doc [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Cleanup references to sample config and redirect users to configuration manual instead.
```
Comment (reivilibre) [inline changelog.d/13077.doc:1]:
```suggestion
Clean up references to sample configuration and redirect users to the configuration manual instead.
```
a wild pedant appears

**Review by reivilibre — CHANGES_REQUESTED** [review]
Thanks for tidying up the loose bits!

File: docs/message_retention_policies.md (hunk 7 above)
Comment (H-Shay) [reply docs/message_retention_policies.md:69]: This particular one is confusing because `retention` is both an option and a section. Essentially, there are sections which generally group functionality together (i.e. server, ratelimiting, etc), then there are options, which is the actual option/flag (ex: `modules`, and then there are sub-options, which modify a given option (ex: `module`). This is all preserved from the original documentation, I just added headers to the sections so they're easier to link to.

**Review by reivilibre — APPROVED** [review]

**Comment by H-Shay** [comment]
Merged.
